### PR TITLE
feat: add a sum filter

### PR DESCRIPTION
**Pull request by EscapedGibbon**
close: https://github.com/image-js/image-js-typescript/issues/320

### CODE REVIEW

File: src/compare/sum.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Image } from '..';
+import checkProcessable from '../utils/validators/checkProcessable';
+import { validateForComparison } from '../utils/validators/validators';
+/**
+ *
+ * Calculate a new image that is the sum between the current image and the otherImage.
+ * @param image - Image to which to add.
+ * @param otherImage - Image to add.
+ * @returns The subtracted image.
```
Comment (stropitek) [inline src/compare/sum.ts:9]: subtracted?

**Review by stropitek — COMMENTED** [review]
Can you add tests which show what happens when the sum saturates (sum exceeds max value). Add a test with 8-bit depth and another with 16-bit depth.

I'm not sure if masks should have a `sum` method. It would have the exact same implementation as the `or` method. @lpatiny @targos should we have `sum` be an alias of `or`?

File: src/compare/sum.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  validateForComparison(image, otherImage);
+
+  const newImage = image.clone();
+  for (let index = 0; index < image.size; index++) {
+    for (let channel = 0; channel < image.channels; channel++) {
+      const value =
+        image.getValueByIndex(index, channel) +
+        otherImage.getValueByIndex(index, channel);
+
+      newImage.setValueByIndex(index, channel, Math.min(value, image.maxValue));
```
Comment (targos) [inline src/compare/sum.ts:29]: Should use `clamp` like in all other methods.

**Comment by targos** [comment]
@stropitek not sure why you're mentioning mask here. The PR doesn't touch anything on mask code.

**Comment by stropitek** [comment]
> not sure why you're mentioning mask here. The PR doesn't touch anything on mask code.

It's because the `subtract` method exists on Mask and to be on par we could implement the `sum` method as well. But to me the concept of doing sums and subtractions on masks seems kind of strange. I don't know for which use case the subtract method exists in the first place.

File: src/compare/add.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * Calculate a new image that is the sum between the current image and the otherImage.
+ * @param image - Image to which to add.
+ * @param otherImage - Image to add.
+ * @returns The summed image.
+ */
+export function add(image: Image, otherImage: Image): Image {
+  if (image instanceof Image) {
+    checkProcessable(image, {
+      bitDepth: [1, 8, 16],
```
Comment (stropitek) [inline src/compare/add.ts:15]: Support for 1-bit depth can be removed

**Comment by stropitek** [comment]
`image.add(otherImage)` should be possible as well

**Review by stropitek — APPROVED** [review]

**Pull request merged**